mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -2776,24 +2776,16 @@ fn electrum_connect(
 }
 
 fn electrum_request(
-    client: &ElectrumClient,
     request: JsonRpcRequestEnum,
     tx: mpsc::Sender<Vec<u8>>,
     responses: JsonRpcPendingRequestsShared,
     timeout: u64,
-) -> Box<dyn Future<Item = JsonRpcResponseEnum, Error = JsonRpcError> + Send + 'static> {
-    let req = request.clone();
-    let client_info = client.coin_name().to_owned();
+) -> Box<dyn Future<Item = JsonRpcResponseEnum, Error = JsonRpcErrorType> + Send + 'static> {
+    let req = request;
     let send_fut = async move {
         let mut json = match json::to_string(&req) {
             Ok(res) => res,
-            Err(err) => {
-                return Err(JsonRpcError::new(
-                    &client_info,
-                    &req,
-                    JsonRpcErrorType::InvalidRequest(err.to_string()),
-                ))
-            },
+            Err(err) => return Err(JsonRpcErrorType::Transport(err.to_string())),
```

Comment:
This is not a Transport Error. This https://github.com/KomodoPlatform/atomicDEX-API/pull/1446#discussion_r961805246 appears to be not fixed.
I am also thinking that we should restrict `electrum_request` to always return `Transport` error somehow to not regress later to returning other `JsonRpcErrorType`, maybe we should have a new error type for this function that maps to `JsonRpcErrorType::Transport` only while writing a comment that this should not be changed above this new error type. What do you think @borngraced ?

Reply:
Actually a simple comment above the `electrum_request` function to always return `JsonRpcErrorType::Transport` is enough. No need for new error types. Just resolve this https://github.com/KomodoPlatform/atomicDEX-API/pull/1446#discussion_r961805246  please.